### DISCUSSION TITLE
About the deployment problem of next js app directory server vercel

### DISCUSSION THREAD

**Discussion by SEOKKAMONI** [vercel/next.js#50794] [Help]
### Summary

I set up a server using the app directory server and connected it from Next and distributed it to Vercel after seeing that it works fine locally, but I got the following error "Application error: a client-side exception has accumulated (see the browser console for more information)." I have to deploy it tomorrow.. Please help me

This is the link to the affected repository >> https://github.com/Bamdoliro/bamdoliro-careers

**Comment by icyJoseph** [comment]
`use client` components cannot be async. 

I think you have a number of issues:

- `use client` components cannot be async, https://github.com/Bamdoliro/bamdoliro-careers/blob/develop/src/app/job/%5Bid%5D/page.tsx
- You must configure styled components register: https://nextjs.org/docs/app/building-your-application/styling/css-in-js#styled-components follow these steps
- You should just get the data at server component level, reaching fro `/api/routes`, is unnecessary and error prone.
   - The things you do here, you should just do in the src/app/job/[id]/page.tsx
   - However, since you use styled components you need to separate the rendering 

```tsx
// src/app/job/[id]/page.tsx
// other imports...
import { JobDetailClient } from './JobDetailClient';

const getJobDetail = async (id: string) => {
 try {
   // make sure to import JOB_DETAIL_DATA
    if (!JOB_DETAIL_DATA[id]) {
      notFound() // import { notFound } from 'next/navigation';
    }
    const { post, ...jobDetail } = JOB_DETAIL_DATA[id];
// better to use import path from 'path';
    const file = fs.readFileSync(path.resolve(process.cwd(), `posts/${post}`), "utf8");

    return {
      markdown: file,
      ...jobDetail,
    };
  } catch {
   notFound();
  }
};

const JobDetail = async ({params}: {params: { id } }) => {
  const props = await getJobDetail(id); // this shouldn't call /api...

  return (
   <JobDetailClient {...props}/>
  );
};

export default JobDetail;
```

And then inside `JobDetailClient` goes your `use client` component, without async and taking the title, etc, props:

```tsx
'use client'

// other imports

export const JobDetailClient = ({ title, markdown, position, googleFormLink }) => {
 // the rest of the file/component
```

 - Your next config is not using styledComponents properly:

```js
module.exports = {
  // other stuff
  compiler: {
    // other compiler options
    styledComponents: true,
  },
}
```

**Reply by icyJoseph** [reply]
Nope, there's no use server, everything is server component first, unless using use client or part of a use client tree.

I've updated the comment above with more stuff

**Reply by SEOKKAMONI** [reply]
thank you !!!!!!

**Reply by devjiwonchoi** [reply]
Addition to @icyJoseph 's answer, I don't think there is an option for `styledComponents = boolean`

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: false,
  styledComponents: true, // seems invalid
};

module.exports = nextConfig;
```

## Reference
- next.config.js Options: https://nextjs.org/docs/app/api-reference/next-config-js

**Reply by icyJoseph** [reply]
Ah right, it is `compiler.styledComponents`: https://nextjs.org/docs/architecture/nextjs-compiler#styled-components

**Reply by icyJoseph** [reply]
@SEOKKAMONI 

I've also fixed this issue:

```tsx
const JobDetail = async ({params}: {params: { id } }) => {
  const props = await getJobDetail(id); 
```

Because `useParams` is only available on client components, but your page is given the params by Next.js
